.github/workflows/SubmitReview.yml

```diff
@@ -0,0 +1,113 @@
+    # PRs from forks don't have required token authorization
+    if: |
+        github.event.pull_request.head.repo.full_name == github.repository
+        && (github.event.review.state == 'approved' || github.event.review.state == 'changes_requested')
+
+    steps:
+      # https://github.com/actions/github-script
+      - uses: actions/github-script@v3
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const TODO_COLUMN = 4971951;
+            const IN_PROGRESS_COLUMN = 4971952;
+            const REVIEW_IN_PROGRESS_COLUMN = 4971953;
+            const REVIEW_APPROVED_COLUMN = 4971954;
+            const DONE_COLUMN = 4971955;
+            //
+            async function getIssue(issue_number) {
+                try {
+                    return (await github.issues.get({
+                        owner: context.repo.owner,
+                        repo: context.repo.repo,
+                        issue_number
+                    })).data;
+                }
+                catch (error) {
+                    console.log(`Issue #${issue_number} not found: ${error}`);
+                    return null;
+                }
+            }
+            //
+            async function findCard(content_url) {
+                // Columns are searched from the most probable one
+                const allColumns = [REVIEW_IN_PROGRESS_COLUMN, REVIEW_APPROVED_COLUMN, IN_PROGRESS_COLUMN, TODO_COLUMN, DONE_COLUMN];
+                for (let i = 0; i < allColumns.length; i++) {
+                    let cards = await github.projects.listCards({ column_id: allColumns[i] });
+                    let card = cards.data.find(x => x.content_url == content_url);
+                    if (card) {
+                        return card;
+                    }
+                }
+                console.log("Card not found for: " + content_url);
+                return null;
+            }
+            //
+            async function removeAssignees(issue){
+                const oldAssignees = issue.assignees.map(x => x.login);
+                if (oldAssignees.length !== 0) {
+                    console.log("Removing assignees: " + oldAssignees.join(", "));
+                    await github.issues.removeAssignees({
+                        owner: context.repo.owner,
+                        repo: context.repo.repo,
+                        issue_number: issue.number,
+                        assignees: oldAssignees
+                    });
+                }
+            }
+            //
+            async function addAssignee(issue, login) {
+                console.log("Assigning to: " + login);
+                await github.issues.addAssignees({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    issue_number: issue.number,
```

Comment:
this is used by `processIssue` to process both `issue` and `PR` - it doesn't seem to change the assignee on a PR, maybe because `issue_number` only works for issues? check [these logs](https://github.com/pavel-mikula-sonarsource/GitHubActionPlayground/runs/1487230650?check_suite_focus=true) for PR: number 3 (no assiggnee on the kanban)

Reply:
I just tried that on the playground and it assigned the issue. 

In GitHub, `PR` is an `Issue` with additional `pull_request` member. Getting [PR #4 from PR API](https://api.github.com/repos/pavel-mikula-sonarsource/GitHubActionPlayground/pulls/4) can be also loaded [from Issues API](https://api.github.com/repos/pavel-mikula-sonarsource/GitHubActionPlayground/issues/4) and both contain `.number`.

API for [Issue #1 from Issue API](https://api.github.com/repos/pavel-mikula-sonarsource/GitHubActionPlayground/issues/1).

I blame GitHub to trick you on this one: 
* PR approval moved the card immediately to `Review approved`. It was not done by our action.
* Action was scheduled and triggered 30 sec later
* You checked the Kanban, card was moved, assignee was missing
* Action started, didn't moved the card, added assignee